### ISSUE TITLE
U+0EBA as Nukta

### ISSUE THREAD

**Issue by Richard57** [MicrosoftDocs/typography-issues#238]
U+0EBA is used as a (consonant) nukta as well as as a vowel killer in one Pali writing system.  In this role, will there by any constraints on its combination with the non-spacing vowel symbols?  It occurs with  U+0EB4, U+0EB5, U+0EB8 and U+0EB9.

The writing system is documented in  Phaya Luang Maha Sena (Phouy)'s, "Learn fast to read Tham characters in Lao Texts [ແບບຣຽນໄວ ເຫຼັ້ມສອງ ຣຽນອ່ານໝັງສືທັມ ຂຽນເປັນພາສາບາລີ]‎", accessible at https://web.archive.org/web/20160401000000*/http://www.laomanuscripts.net/downloads/tham_pali.pdf, starting at p40.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: fc1eb7c3-0ea8-2fec-5ebf-12d25736c619
* Version Independent ID: 33cd23ee-8352-9d19-d6f8-69c37758f70b
* Content: [Developing OpenType Fonts for Lao Script - Typography](https://docs.microsoft.com/en-us/typography/script-development/lao)
* Content Source: [typographydocs/script-development/lao.md](https://github.com/MicrosoftDocs/typography/blob/live/typographydocs/script-development/lao.md)
* Product: **typography**
* GitHub Login: @mijacobs
* Microsoft Alias: **mijacobs**

**Comment by xadxura** [comment]
ThaiLao engine. Assigning to @PeterCon

**Comment by behdad** [comment]
Do we need any changes in HarfBuzz? cc @dscobertt

**Comment by dscorbett** [comment]
HarfBuzz doesn’t need any changes to support this character.

HarfBuzz imposes no cooccurrence or ordering constraints on Lao marks, so U+0EBA may safely be used as a nukta. [Line 7 of page 45 of the above-linked document](https://web.archive.org/web/20160429020633/http://www.laomanuscripts.net/downloads/tham_pali.pdf) shows the syllable ⟨ຍ຺ູ⟩ with the nukta to the left of the vowel sign, indicating that the NFC form (i.e. nukta before vowel sign) is fine for Lao. (For Thai, HarfBuzz rearranges some combining character classes such that the cognate U+0E3A THAI CHARACTER PHINTHU appears under below-base vowels. According to [case 3 of L2/18-216](https://www.unicode.org/L2/L2018/18216-thai-order.pdf#page=6), this is wrong.)

**Comment by ohbendy** [comment]
Yes, I believe this is a reordering bug in Harfbuzz too. Oddly, Uniscribe appears to reorder the phinthu 0E3A after any belowvowels, contrary to the combining classes. In all the orthographies I've seen (for Akha, Gong, Lawa and Pattani Malay languages), the phinthu should attach to the base, and any belowvowels should stack below that. In other orthographies that use phinthu (Khmer Surin, Mien, Mlabri, Pali-Sanskrit), it does not co-occur with belowvowels.
